### PR TITLE
ICICI Data API

### PR DESCRIPTION
**Pull request by psymbio**
Function for fetching data from ICICI Breeze API is added in: `broker/icici/api/data.py` and the call is changed from Fyers to ICICI here: `restx_api/history.py`, however, I believe this needs to be done in such a way that the switching between different brokers is handled in this file.

For testing I have added: `strategies/testing_icici_data_api.ipynb` but in this you'll be able to see the data is fetched but with a processing problem:

```bash
{'status': 'error',
 'message': "Failed to process historical data: 'timestamp'",
 'error_type': 'processing_error',
 'raw_data': [{'close': 242.4,
   'datetime': '2024-12-20 11:22:00',
   'high': 242.4,
   'low': 242.2,
   'open': 242.2,
   'volume': 16220},
```

wasn't sure exactly where in the code this is done but it's just that instead of timestamp ICICI calls it datetime.

### CODE REVIEW

**Comment by psymbio** [comment]
Additional features to be added:

1. product_type: This will not always be "Cash" and needs to be updated as a string variable with possible values ("Futures", "Options", "Cash"). For simplicity, it could default to "Cash".
2. expiry_date: A string representing the expiration date.
3. right: A string variable to specify the type ("call", "put", or "others").
4. strike_price: A string to represent the strike price.

Based on: https://api.icicidirect.com/breezeapi/documents/index.html#gethistoricalchartslist

**Comment by marketcalls** [comment]
Hi Recently added Added AngelAPI and removed the Hardcoded Class FyersData from the restxapi and renamed the class to BrokerData. Can you please refer and change the Class Name. I will test and approve the pr accordingly.

File: broker/icici/api/data.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        'Authorization': f'Bearer {AUTH_TOKEN}',
+        'Content-Type': 'application/json'
+    }
+
+    conn.request(method, endpoint, payload, headers)
+    res = conn.getresponse()
+    data = res.read()
+    return json.loads(data.decode("utf-8"))
+
+class ICICIDirectData:
```
Comment (crypt0inf0) [inline broker/icici/api/data.py]: Rename `ICICIDirectData` to `BrokerData` for common class name.

File: restx_api/history.py [hunk 2]
```diff
@@ -56,7 +56,8 @@ def post(self):
 
             try:
                 # Initialize broker's data handler
-                data_handler = broker_module.FyersData(AUTH_TOKEN)
+                # data_handler = broker_module.FyersData(AUTH_TOKEN)
+                data_handler = broker_module.ICICIDirectData(AUTH_TOKEN)
```
Comment (crypt0inf0) [inline restx_api/history.py]: Rename `ICICIDirectData` to `BrokerData`

**Pull request closed without merging**